paper-server/patches/sources/net/minecraft/world/item/EnderpearlItem.java.patch

```diff
@@ -39,7 +39,11 @@
 +                );
 +                player.awardStat(Stats.ITEM_USED.get(this));
 +            } else {
-+            // Paper end - PlayerLaunchProjectileEvent
++                if (player instanceof net.minecraft.server.level.ServerPlayer serverPlayer) {
++                    serverPlayer.deregisterEnderPearl(thrownEnderpearl.projectile());
++                    serverPlayer.connection.send(new net.minecraft.network.protocol.game.ClientboundCooldownPacket(player.getCooldowns().getCooldownGroup(itemInHand), 0)); // prevent visual desync of cooldown on the slot
```

Comment:
just for curious this is the same than `serverPlayer.getCooldowns().addCooldown(player.getCooldowns().getCooldownGroup(itemInHand), 0, false);` ? well just the map thing.

Reply:
Not wanting to store it in the map is defo intentionally, basically we want to revert the client state without messing with the server so we want to avoid using that.